### ISSUE TITLE
Validator ignoring 'always=True' on optional fields on inheritance

### ISSUE THREAD

**Issue by gaja-dev** [samuelcolvin/pydantic#1155]
# Possible Bug

**Validator seems to ignore the 'always' parameter on inheritance since pydantic >= 1.2**

I had a running code that uses inheritance on the models with an enforced field that is used to decide if other optional fields will be checked, or not. This worked well until I updated to pydantic 1.3. Further testing shows that the code fails since pydantic 1.2. So there seem to be some changes from 1.1 to 1.2 that leads to this issue.

Unfortunately I could not find a previous issue, or documentation hint that explains this deviating behavior to me.

If I missed something in the doc, please give me a hint and excuse my issue. Otherwise, I would be thankful if someone could fix this, or explain how the intended behavior can be achieved on pydantic >=1.2

Please find an example code and corresponding output below.

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
pydantic version: 1.3
pydantic compiled: False
install path: [...]\python_pydantic-ni4Wh3vY\Lib\site-packages\pydantic
python version: 3.6.8 (tags/v3.6.8:3c6b436a57, Dec 24 2018, 00:16:47) [MSC v.1916 64 bit (AMD64)]
platform: Windows-10-10.0.18362-SP0
optional deps. installed: []
```

Here is an example code to reproduce this issue and below that, the corresponding output on pydantic 1.1 and >=1.2:
```py
import typing

import pydantic

class A(pydantic.BaseModel):

    must_validate: bool

    class Config:
        extra = pydantic.Extra.ignore


class B(pydantic.BaseModel):

    value_a: typing.Optional[str] = None
    value_b: typing.Optional[str] = None

    class Config:
        extra = pydantic.Extra.ignore

class C(B, A):

    @pydantic.validator("value_a", "value_b", always=True)
    def validate_if_enforced(cls, value, values):
        if "must_validate" in values and values["must_validate"]:
            if not value:
                raise ValueError("must be set if 'must_validate' is True")
        return value
    
    class Config:
        extra = pydantic.Extra.forbid


if __name__ == "__main__":
    success_data = {
        "must_validate": True,
        "value_a": "foo",
        "value_b": "bar",
    }

    fail_data = {
        "must_validate": True,
        "value_a": "foo",
    }

    try:
        success = C.parse_obj(success_data)
        print(success.dict())

        fail = C.parse_obj(fail_data)
        print(fail.dict())
    except pydantic.ValidationError as exc:
        print(exc)
...
```

Output on **pydantic 1.1** (This is the intended behavior!)
```
{'must_validate': True, 'value_a': 'foo', 'value_b': 'bar'}
1 validation error for C
value_b
  must be set if 'must_validate' is True (type=value_error)
```

Output on **pydantic >=1.2**
```
{'must_validate': True, 'value_a': 'foo', 'value_b': 'bar'}
{'must_validate': True, 'value_a': 'foo', 'value_b': None}
```

**Comment by samuelcolvin** [comment]
Thanks for reporting, I get the same.

The intermediate model `B` is not required to produce the error.

The problem is b521f6b24aa48ca65044d3a5a693ad44d929307d and almost certainly the change `main.py:196`.

It might be as simple as reversing that change, but I made it for a reason (that currently escapes me). I'll look at it next week or when I get a chance, if it's urgent feel free to investigate and submit a PR if you can find a fix.

**Comment by PrettyWood** [comment]
Fixed by #1545
